tests/scalers/nats/nats_jetstream/nats_jetstream_test.go

```diff
@@ -0,0 +1,206 @@
+//go:build e2e
+// +build e2e
+
+package jetstream_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	k8s "k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+	nats "github.com/kedacore/keda/v2/tests/scalers/nats/helper"
+)
+
+// Load env variables from .env files
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "nats-jetstream-test"
+)
+
+var (
+	testNamespace                = fmt.Sprintf("%s-ns", testName)
+	natsNamespace                = fmt.Sprintf("%s-nats-ns", testName)
+	natsAddress                  = fmt.Sprintf("nats://nats.%s.svc.cluster.local:4222", natsNamespace)
+	natsServerMonitoringEndpoint = fmt.Sprintf("nats.%s.svc.cluster.local:8222", natsNamespace)
+	messagePublishCount          = 1000
+	deploymentName               = "sub"
+	minReplicaCount              = 0
+	maxReplicaCount              = 2
+)
+
+type templateData struct {
+	TestNamespace                string
+	NatsAddress                  string
+	NatsServerMonitoringEndpoint string
+	NumberOfMessages             int
+}
+
+type templateValues map[string]string
+
+const (
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sub
+  namespace: {{.TestNamespace}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: sub
+  template:
+    metadata:
+      labels:
+        app: sub
+    spec:
+      containers:
+      - name: sub
+        image: "goku321/nats-consumer:v0.9"
```

Comment:
could you please add source code for this image here: https://github.com/kedacore/test-tools/tree/main/e2e/images ?

Reply:
(This could be done after this PR is merged)